app/src/main/java/com/example/user/my_calculator/MainActivity.java

```diff
@@ -12,290 +12,132 @@
-            @SuppressLint("SetTextI18n")
-            @Override
-            public void onClick(View v) {
-                edtInput.setText(edtInput.getText() + "7");
-                edtInput.setSelection(edtInput.getText().length());
-            }
-        });
+    @SuppressLint("SetTextI18n")
+    public void onSubClick(View view) {
+        action = "-";
+        if (!Double.isNaN(firstNumber)) {
+            secondNumber = Double.parseDouble(edtInput.getText().toString());
+            firstNumber = firstNumber - secondNumber;
+        } else {
+            firstNumber = Double.parseDouble(edtInput.getText().toString());
+        }
+
+        txtResult.setText(firstNumber + " - ");
+        edtInput.setText(null);
+    }
 
-        btnNumber8.setOnClickListener(new View.OnClickListener() {
-            @SuppressLint("SetTextI18n")
-            @Override
-            public void onClick(View v) {
-                edtInput.setText(edtInput.getText() + "8");
-                edtInput.setSelection(edtInput.getText().length());
-            }
-        });
+    @SuppressLint("SetTextI18n")
+    public void onMulClick(View view) {
+        action = "*";
+        if (!Double.isNaN(firstNumber)) {
+            secondNumber = Double.parseDouble(edtInput.getText().toString());
+            firstNumber = firstNumber * secondNumber;
+        } else {
+            firstNumber = Double.parseDouble(edtInput.getText().toString());
+        }
+
+        txtResult.setText(firstNumber + " * ");
+        edtInput.setText(null);
+    }
 
-        btnNumber9.setOnClickListener(new View.OnClickListener() {
-            @SuppressLint("SetTextI18n")
-            @Override
-            public void onClick(View v) {
-                edtInput.setText(edtInput.getText() + "9");
-                edtInput.setSelection(edtInput.getText().length());
-            }
-        });
+    @SuppressLint("SetTextI18n")
+    public void onDotClick(View view) {
+        edtInput.setText(edtInput.getText() + ".");
+        edtInput.setSelection(edtInput.getText().length());
+    }
 
-        btnDot.setOnClickListener(new View.OnClickListener() {
-            @SuppressLint("SetTextI18n")
-            @Override
-            public void onClick(View v) {
-                edtInput.setText(edtInput.getText() + ".");
-                edtInput.setSelection(edtInput.getText().length());
-            }
-        });
+    @SuppressLint("SetTextI18n")
+    public void onDivClick(View view) {
+        action = "/";
+        if (!Double.isNaN(firstNumber)) {
+            secondNumber = Double.parseDouble(edtInput.getText().toString());
+            firstNumber = firstNumber / secondNumber;
+        } else {
+            firstNumber = Double.parseDouble(edtInput.getText().toString());
+        }
+
+        txtResult.setText(firstNumber + " / ");
+        edtInput.setText(null);
+    }
 
+    public void onResultClick(View view) {
+        if (action != null && action.equals("+")) {
```

Comment:
action != null && action.equals("+") could be simplified to "+".equals(action) also please move all strings to constant